src/qz/utils/CommandParser.java

```diff
@@ -0,0 +1,220 @@
+/**
+ * @author Tres Finocchiaro
+ *
+ * Copyright (C) 2019 Tres Finocchiaro, QZ Industries, LLC
+ *
+ * LGPL 2.1 This is free software.  This software and source code are released under
+ * the "LGPL 2.1 License".  A copy of this license should be distributed with
+ * this software. http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+
+package qz.utils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import qz.common.Constants;
+import qz.common.SecurityInfo;
+import qz.exception.NullCommandException;
+import qz.installer.Installer;
+import qz.installer.TaskKiller;
+import qz.installer.certificate.PropertiesLoader;
+
+import java.io.File;
+import java.util.*;
+
+import static qz.common.Constants.*;
+import static qz.utils.CommandParser.ExitStatus.*;
+
+public class CommandParser {
+    public enum ExitStatus {
+        SUCCESS(0),
+        GENERAL_ERROR(1),
+        USAGE_ERROR(2),
+        NO_AUTOSTART(3);
+        private int code;
+        ExitStatus(int code) {
+            this.code = code;
+        }
+        public int getCode() {
+            return code;
+        }
+    }
+
+    protected static final Logger log = LoggerFactory.getLogger(CommandParser.class);
+    private List<String> args;
+    private ExitStatus exitStatus;
+
+    public CommandParser(String[] args) {
+        this.exitStatus = SUCCESS;
+        this.args = new ArrayList<>(Arrays.asList(args));
+    }
+    public List<String> getArgs() {
+        return args;
+    }
+
+    public ExitStatus getExitStatus() {
+        return exitStatus;
+    }
+
+    public int getExitCode() {
+        return exitStatus.getCode();
+    }
+
+    /**
+     * Gets the requested flag status
+     */
+    public boolean argFlag(String ... matches) {
```

Comment:
Similar for this, could call this "hasFlag" or "isSet", would be clearer to its purpose too

Reply:
changed to `hasFlag` per recommendation.